### PR TITLE
test_: set proxy creds

### LINKED ISSUE #6144: Set proxy credentials from functional tests
1. Read these env vars:
    - `STATUS_BUILD_PROXY_USER `
    - `STATUS_BUILD_PROXY_PASSWORD `

2. Pass values to these endpoints as part of `WalletSecretsConfig`:
    - `CreateAccountAndLogin`
    - `RestoreAccountAndLogin`
    - `LoginAccount` 
    
    https://github.com/status-im/status-go/blob/29bf0d6c8e34bc3663a3ed9678061a8b215238db/params/config.go#L551-L556

    Reference status-desktop for correspondence of env vars to parameters:
    https://github.com/status-im/status-desktop/blob/73af1b8ba0fd82a5259a5799fef3bd03280869b3/src/app_service/service/accounts/service.nim#L168-L172

    ```nim
      proc buildWalletSecrets(): WalletSecretsConfig =
        return WalletSecretsConfig(
          # ...
          statusProxyMarketUser: STATUS_PROXY_USER_RESOLVED,
          statusProxyMarketPassword: STATUS_PROXY_PASSWORD_RESOLVED,
          statusProxyBlockchainUser: STATUS_PROXY_USER_RESOLVED,
          statusProxyBlockchainPassword: STATUS_PROXY_PASSWORD_RESOLVED,
        )
    ```

3.  Also set `WalletSecretsConfig.statusProxyStageName` to `"test"`.

### PR DESCRIPTION
**Pull request by antdanchenko**
Closes https://github.com/status-im/status-go/issues/6144

### CODE REVIEW

**Review by anastasiyaig — APPROVED** [review]

File: tests-functional/clients/status_backend.py [hunk 1]
```diff
@@ -133,6 +133,20 @@ def init_status_backend(self, data_dir=USER_DIR):
...
+            password = os.environ["STATUS_BUILD_PROXY_PASSWORD"]
+
+            data["StatusProxyMarketUser"] = user
+            data["StatusProxyMarketPassword"] = password
+            data["StatusProxyBlockchainUser"] = user
+            data["StatusProxyBlockchainPassword"] = password
+
+            data["StatusProxyEnabled"] = True
+            data["StatusProxyStageName"] = "test"
+        return data
```
Comment (igor-sirotin) [inline tests-functional/clients/status_backend.py]: Just a suggestion for early return pattern:
```suggestion
        if not "STATUS_BUILD_PROXY_USER" in os.environ:
            return data

        user = os.environ["STATUS_BUILD_PROXY_USER"]
        password = os.environ["STATUS_BUILD_PROXY_PASSWORD"]

        data["StatusProxyMarketUser"] = user
        data["StatusProxyMarketPassword"] = password
        data["StatusProxyBlockchainUser"] = user
        data["StatusProxyBlockchainPassword"] = password

        data["StatusProxyEnabled"] = True
        data["StatusProxyStageName"] = "test"
        return data
```

**Review by igor-sirotin — COMMENTED** [review]
@antdanchenko this also needs to be set in `LoginAccount`:
https://github.com/status-im/status-go/blob/1795620df0e93cc78be04e038e40ee589ec462f2/tests-functional/clients/status_backend.py#L180-L187 https://github.com/status-im/status-go/blob/b74d9e6b4e9177f1aa9ba57f9de8beebf5b352e1/protocol/requests/login.go#L36

**Review by igor-sirotin — APPROVED** [review]
Thank you!

File: _assets/ci/Jenkinsfile.tests-rpc [hunk 2]
```diff
@@ -47,6 +47,11 @@ pipeline {
             credentialsId: 'codecov-repository-upload-token',
             variable: 'CODECOV_TOKEN'
           ),
+          usernamePassword(
+            credentialsId: "api-proxy-devel",
+            usernameVariable: 'STATUS_BUILD_PROXY_USER',
+            passwordVariable: 'STATUS_BUILD_PROXY_PASSWORD'
+    ),
```
Comment (jakubgs) [inline _assets/ci/Jenkinsfile.tests-rpc:54]: Why are we using credentials for development builds for tests? We have credentials for that:
https://github.com/status-im/infra-proxy/blob/4591fae365169eefae843caafb37afd478ddc07c/ansible/group_vars/proxy.test.yml#L16

Comment (igor-sirotin) [reply _assets/ci/Jenkinsfile.tests-rpc:54]: @jakubgs I think it was suggested here 🤔 
https://github.com/status-im/status-go/issues/6145#issuecomment-2517803285

Comment (mendelskiv93) [reply _assets/ci/Jenkinsfile.tests-rpc:54]: Credentials being used are indeed the `proxy.test` ones linked [here](https://github.com/status-im/infra-proxy/blob/4591fae365169eefae843caafb37afd478ddc07c/ansible/group_vars/proxy.test.yml#L16). The `credentialsId` naming could ideally be `api-proxy-test`, but I kept `devel` consistent across all projects. i'll update the ID or add a comment to clarify and avoid confusion in the future

**Pull request merged**